src/ResourceManager/AzureBatch/Commands.Batch/Models/BatchClient.Files.cs

```diff
@@ -219,12 +222,35 @@ public void DownloadNodeFile(DownloadNodeFileOptions options)
                     }
             }
 
-            DownloadNodeFileByInstance(nodeFile, options.DestinationPath, options.Stream, options.AdditionalBehaviors);
+            DownloadNodeFileByInstance(nodeFile, options.DestinationPath, options.Stream, options.RangeStart, options.RangeEnd, options.AdditionalBehaviors);
         }
 
         // Downloads the file represented by an NodeFile instance to the specified path.
-        private void DownloadNodeFileByInstance(NodeFile file, string destinationPath, Stream stream, IEnumerable<BatchClientBehavior> additionalBehaviors = null)
+        private void DownloadNodeFileByInstance(NodeFile file, string destinationPath, Stream stream, long? rangeStart, long? rangeEnd, IEnumerable<BatchClientBehavior> additionalBehaviors = null)
         {
+            Protocol.RequestInterceptor interceptor = new Protocol.RequestInterceptor(baseRequest =>
+            {
+                var fromTaskRequest = baseRequest as BatchRequests.FileGetFromTaskBatchRequest;
+                if (fromTaskRequest != null && rangeStart != null && rangeEnd != null)
```

Comment:
Yeah it would behave correctly, but maybe should make it clearer that that's the case.

Basically the defaults work such that they can either both be set or neither is set. In the case where neither is set we do what we did before (don't include the range header). In the case where one is set, the other is defaulted (to `0` for start or to `long.Max` for End)